### PR TITLE
DEVX-2301: Update validation requirements

### PR DESCRIPTION
**Pull request by AmyJZhao**
## Description

`install.yml` configuration was updated, and new validation requirements were added.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/guides/index.md [hunk 1]
```diff
@@ -33,9 +35,13 @@ So you've just created a new template for App Builder and want to make it availa
...
-- If the template contains the `services` property to declare the services it uses, `services` must be an array of objects representing a service the template depends on. Each object must contain the `code` property. `services` is an optional property.
+| Field        | Type | Description                                                                                                                    | Required           |
+| ------------ | ------------ | ------------------------------------------------------------------------------------------------------------------------------ | ------------------ |
+| `categories` | list of strings | Template must have categories defined.                                                                                          | Yes |
+| `extension`  | object | Template might implement an extension point. One of its properties must be `serviceCode`, which will contain the extension metadata. | No |
+| `apis`   | list of objects | List the APIs required to be added to all Workspaces in the App Builder Project, if they don't exist. Each API object must contain `code` as one of its properties, this is the sdk code of the service. | No |
```
Comment (roribio) [inline src/pages/guides/index.md:42]: This filed appears as `apis` and the example still lists `services`. 


Please also see this proposal for the `install.yaml` schema: https://git.corp.adobe.com/devx-services/aio-lib-console-project-installation/pull/1/files#diff-5f9599c1addcd4716b953264f83abee501e3588d24cf1242f906a0536f6c040a

File: src/pages/guides/index.md [hunk 2]
```diff
@@ -49,4 +55,17 @@ services:
   - code: CampaignStandard
 extension:
   serviceCode: dx/excshell/1
+runtime: true
```
Comment (roribio) [inline src/pages/guides/index.md:58]: `runtime` and `event` are marked as future improvements in this document: https://wiki.corp.adobe.com/pages/viewpage.action?spaceKey=AdobeCloudPlatform&title=App+Builder+Template+Install+Configuration.

I think they should be out of scope for now. @slitviachenko thoughts?

[the remark this answers is not in the document]
Comment (slitviachenko) [reply src/pages/guides/index.md:58]: @roribio , I see in the "Specification" section that only "hooks" is marked as "Future", "runtime" and "event" are new properties. Shaz also mentioned those changes in [his comment](https://adobedx.slack.com/archives/C02UWB27XJN/p1653487808846049). So I think we should support them.

Comment (slitviachenko) [reply src/pages/guides/index.md:58]: @roribio , I see in the "Specification" section that only "hooks" is marked as "Future", "runtime" and "event" are new properties. Shaz also mentioned those changes in [his comment](https://adobedx.slack.com/archives/C02UWB27XJN/p1653487808846049). So I think we should support them.

**Review by slitviachenko — CHANGES_REQUESTED** [review]
@AmyJZhao , please update `src/openapi.yaml` as well

**Review by slitviachenko — APPROVED** [review]

File: src/pages/guides/index.md (hunk 1 above)
Comment (AmyJZhao) [reply src/pages/guides/index.md:42]: See @slitviachenko's comment below

**Pull request merged**